applications/dashboard/models/class.rolemodel.php

```diff
@@ -750,4 +750,41 @@ public function deleteAndReplace($roleID, $newRoleID) {
         $result = $this->SQL->delete('Role', array('RoleID' => $roleID));
         return $result;
     }
+
+    /**
+     * Get a list of a user's roles, optionally return all the role data or just one field name.
+     *
+     * @param $userID
+     * @param string $field optionally the field name from the role table to return.
+     * @return array|null|void
+     */
+    public function getUserRoles($userID, $field="Name") {
```

Comment:
this isn't a very good method name, because we're filtering by permission. the name suggests it will give you ALL roles.